### PR TITLE
feat(abci): Utilize geth `miner` to build blocks

### PR DESCRIPTION
**Pull request by itsdevbear**
### Summary by CodeRabbit

- New Feature: Introduced a new "miner" package that implements Ethereum mining operations, enhancing the blockchain's functionality.
- Refactor: Updated function calls within the `Polaris` struct to use the new `spminer` object, reflecting a change in the mining mechanism.
- Refactor: Renamed the `Broadcaster` interface to `TxBroadcaster` and updated related functions and structures accordingly for better clarity.
- New Feature: Added new functions to the `Polaris` struct including `SPMiner()`, `MinerChain()`, and `IsLocalBlock()` to enhance block management capabilities.
- Bug Fix: Modified JSON responses for the `eth_getBlockByNumber` RPC call to correctly represent data size and withdrawals.
- Refactor: Shifted from using a blockchain object to a miner object for operations in `backend.go`, improving code organization.
- Refactor: Replaced references to `app.InterfaceRegistry()` with `app.interfaceRegistry` and `ctx.KVStore` with `ctx.UnsafeFindStoreKey` for improved code efficiency.

### CODE REVIEW

File: cosmos/txpool/mempool.go [hunk 1]
```diff
@@ -39,14 +45,21 @@ var _ mempool.Mempool = (*Mempool)(nil)
 type GethTxPool interface {
 	Add([]*coretypes.Transaction, bool, bool) []error
 	Stats() (int, int)
+	SubscribeNewTxsEvent(chan<- core.NewTxsEvent) event.Subscription
+}
+
+// Startable represents a type that can be started.
+type Startable interface {
```
Comment (calbera) [inline cosmos/txpool/mempool.go:52]: move into the lib maybe

File: cosmos/miner/miner.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+// Miner implements the baseapp.TxSelector interface.
+type Miner struct {
+	*miner.Miner
+	serializer     evmtypes.TxSerializer
+	currentPayload *miner.Payload
+}
+
+// New produces a cosmos miner from a geth miner.
+func New(gm *miner.Miner) *Miner {
```
Comment (calbera) [inline cosmos/miner/miner.go:49]: gm gm

File: eth/polar/polaris.go [hunk 3]
```diff
@@ -101,6 +107,7 @@ func NewWithNetworkingStack(
 		stack:      stack,
 		host:       host,
 		engine:     host.GetEnginePlugin(),
+		beacon:     &consensus.MockEngine{},
```
Comment (calbera) [inline eth/polar/polaris.go:110]: honestly can rename (to not include "mock") if this is actually being used as beacon

File: eth/miner/miner.go [hunk 4]
```diff
@@ -65,6 +65,7 @@ type Miner interface {
 }
 
 // miner implements the Miner interface.
+// TODO: RENAME TO STATE PROCESSOR OR DEPRECATE.
```
Comment (calbera) [inline eth/miner/miner.go:68]: deprecate would be fire but not convinced that we are setting everything up correct with just the geth one

Comment (itsdevbear) [reply eth/miner/miner.go:68]: lifecycle management in general needs to be improved, later pr tho tbh

File: cosmos/txpool/mempool.go (hunk 1 above)
Comment (calbera) [reply cosmos/txpool/mempool.go:52]: True but go does it a bit too. Check out the default built in `io.Closer` type

**Comment by itsdevbear** [comment]
@ocnc @calbera @BrickBera let's get this guy on devnet by EOD if possible 😤😤😤

**Review by calbera — APPROVED** [review]

**Pull request merged**